### PR TITLE
Ble sample

### PR DESCRIPTION
**Pull request by jperedadnr**

### CODE REVIEW

File: beacons/pom.xml [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    <name>beacons</name>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <maven.compiler.release>11</maven.compiler.release>
+        <javafx.version>14.0.1</javafx.version>
+        <charm.version>6.0.4-SNAPSHOT</charm.version>
+        <attach.version>4.0.7</attach.version>
+        <javafx-maven-plugin-version>0.0.4</javafx-maven-plugin-version>
+        <client-maven-plugin-version>0.1.22-SNAPSHOT</client-maven-plugin-version>
```
Comment (abhinayagarwal) [inline beacons/pom.xml:19]: Why do we still need the snapshot version to run this sample?

File: beacons/pom.xml [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+                    <target>${clientTarget}</target>
+                    <attachList>
+                        <list>ble</list>
+                        <list>display</list>
+                        <list>lifecycle</list>
+                        <list>statusbar</list>
+                        <list>storage</list>
+                    </attachList>
+                    <reflectionList>
+                        <list>javafx.fxml.FXMLLoader</list>
```
Comment (abhinayagarwal) [inline beacons/pom.xml:120]: This should not be required any longer

File: beacons/src/android/AndroidManifest.xml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    <application a:label='beacons' a:icon="@mipmap/ic_launcher">
+        <activity a:name='com.gluonhq.helloandroid.MainActivity' a:configChanges="orientation|keyboardHidden">
+            <intent-filter>
+                <category a:name='android.intent.category.LAUNCHER'/>
+                <action a:name='android.intent.action.MAIN'/>
+            </intent-filter>
+        </activity>
+        <activity a:name='com.gluonhq.helloandroid.PermissionRequestActivity'/>
+    </application>
+</manifest>
```
Comment (abhinayagarwal) [inline beacons/src/android/AndroidManifest.xml:20]: Do we need the manifest file because we need permission for using Bluetooth services?

File: beacons/src/ios/Default-Info.plist [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+	<key>UIRequiresFullScreen</key>
+	<string>YES</string>
+    <key>NSLocationWhenInUseUsageDescription</key>
+    <string>This app needs location services</string>
+    <key>NSLocationAlwaysAndWhenInUseUsageDescription</key>
+    <string>This app needs location services</string>
+    <key>NSBluetoothAlwaysUsageDescription</key>
+    <string>This app needs bluetooth services</string>
+</dict>
+</plist>
```
Comment (abhinayagarwal) [inline beacons/src/ios/Default-Info.plist:55]: Same goes for this file. Is it required for enabling bluetooth services?

File: beacons/src/ios/assets/Assets.xcassets/AppIcon.appiconset/Contents.json [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        "size" : "1024x1024",
+        "filename" : "Gluon-app-store-icon-1024@1x.png",
+        "scale" : "1x"
+        }
+    ],
+    "info" : {
+        "version" : 1,
+        "author" : "xcode"
+    }
+}
```
Comment (abhinayagarwal) [inline beacons/src/ios/assets/Assets.xcassets/AppIcon.appiconset/Contents.json:116]: Is this file required? Also, do we need all the images for iOS and Android?

File: beacons/pom.xml (hunk 1 above)
Comment (jperedadnr) [reply beacons/pom.xml:19]: There are important changes merged after the last release.

File: beacons/pom.xml (hunk 2 above)
Comment (jperedadnr) [reply beacons/pom.xml:120]: true, (but only if we use the snapshot). Removed

File: beacons/pom.xml [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+    <version>1.0.0-SNAPSHOT</version>
+    <packaging>jar</packaging>
+
+    <name>beacons</name>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <maven.compiler.release>11</maven.compiler.release>
+        <javafx.version>14.0.1</javafx.version>
+        <charm.version>6.0.4-SNAPSHOT</charm.version>
```
Comment (abhinayagarwal) [inline beacons/pom.xml:16]: This can be updated to the release version

Comment (jperedadnr) [reply beacons/pom.xml]: Done

File: beacons/pom.xml (hunk 1 above)
Comment (abhinayagarwal) [reply beacons/pom.xml]: This can be updated now.

Comment (abhinayagarwal) [reply beacons/pom.xml]: Ok, I see this has been updated. Marking the conversation as "Resolved".

File: beacons/src/android/AndroidManifest.xml (hunk 3 above)
Comment (abhinayagarwal) [reply beacons/src/android/AndroidManifest.xml:20]: Can this be marked as resolved?

Comment (jperedadnr) [reply beacons/src/android/AndroidManifest.xml:20]: The issue has been solved (no need to copy the assets, only the manifest), but will depend on 1.24-SNAPSHOT, or a release of 1.24

Comment (abhinayagarwal) [reply beacons/src/android/AndroidManifest.xml:20]: Does this mean that most of the assets in this PR can be removed?

Comment (jperedadnr) [reply beacons/src/android/AndroidManifest.xml:20]: All of the iOS/Android assets can be removed, yes. Only if we modify the plugin (waiting for a release...)

Comment (jperedadnr) [reply beacons/src/android/AndroidManifest.xml:20]: Ok, assets removed, pom updated

**Review by abhinayagarwal — APPROVED** [review]

File: beacons/src/android/AndroidManifest.xml (hunk 3 above)
Comment (jperedadnr) [reply beacons/src/android/AndroidManifest.xml:20]: The manifest file can be removed now, permissions are handled by Attach.
